src/uu/split/src/split.rs

```diff
@@ -288,6 +301,76 @@ fn num_prefix(i: usize, width: usize) -> String {
     c
 }
 
+/// A writer that writes to a shell_process' stdin
+struct FilterWriter {
+    /// Running shell process
+    shell_process: Child,
+}
+
+impl Write for FilterWriter {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        self.shell_process
+            .stdin
+            .as_mut()
+            .expect("failed to get shell stdin")
+            .write(buf)
+    }
+    fn flush(&mut self) -> Result<()> {
+        self.shell_process
```

Comment:
codecov complains here, I guess it is too hard to test?

Reply:
I can `flush()` in `FilterWriter.drop()`